packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -293,6 +310,81 @@ describe('ExpoSQLiteDatabase 2', () => {
 				)
 			).rejects.toMatchObject(MOCK_SQL_ERROR);
 		});
+
+		it('should reject promise and return error when the transaction is unsuccessful', async () => {
+			mockTransaction.executeSql
+				.mockImplementationOnce(
+					(statement, params, successCallback, errorCallback) => {
+						errorCallback(mockTransaction, MOCK_SQL_ERROR);
+					}
+				)
+				.mockImplementation(
+					(statement, params, successCallback, errorCallback) => {
+						successCallback(mockTransaction, MOCK_RESULT_SET_STRING);
+					}
+				);
+
+			await expect(
+				instance.batchSave(
+					saveParameterizedStatements,
+					deleteParameterizedStatements
+				)
+			).rejects.toStrictEqual(MOCK_SQL_ERROR);
+		});
+
+		// test sqlerror in when running deleteStatements
+		it('should reject promise and return error when the transaction is unsuccessful', async () => {
```

Comment:
Try to incorporate the condition into the test description rather than relying on comment - this way the test runner printout is meaningful. Again, we should never have multiple tests with the same description (and this description shows up quite a few times 😓 ) Let's try to distinguish them by describing what the test is doing differently

Reply:
Will update that.